build.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow::Result;
+
+fn detect_fedora_feature() -> Result<()> {
+    if !std::path::Path::new("/usr/lib/os-release").exists() {
+        return Ok(());
+    }
+    let p = std::process::Command::new("bash")
+        .args(&["-c", ". /usr/lib/os-release && echo ${ID}"])
+        .stdout(std::process::Stdio::piped())
+        .output()?;
+    let out = std::str::from_utf8(&p.stdout).ok().map(|s| s.trim());
+    if out == Some("fedora") {
+        println!("cargo:rustc-cfg=feature=\"fedora-integration\"")
+    }
+    Ok(())
+}
+
+fn main() -> Result<()> {
```

Comment:
Note this introduction of `build.rs` is something we'll use in the future for sure too.